contracts/ExpectedRate.sol

```diff
@@ -54,10 +54,27 @@ contract ExpectedRate is Withdrawable, ExpectedRateInterface, Utils2 {
 
         if (srcQty == 0) srcQty = 1;
 
-        bool didRevert = false;
+        //not needed
+        uint bestReserve;
 
-        (didRevert, expectedRate, slippageRate) = safeFindBestRate(src, dest, srcQty, usePermissionless);
-        if (didRevert) return (0, 0);
+        if (usePermissionless) {
+            (bestReserve, expectedRate) = kyberNetwork.findBestRate(src, dest, srcQty);
+
+            if (quantityFactor != 1) {
+                (bestReserve, slippageRate) = kyberNetwork.findBestRate(src, dest, (srcQty * quantityFactor));
+            } else {
+                slippageRate = expectedRate;
+            }
+        } else {
+            (bestReserve, expectedRate) = kyberNetwork.findBestRateOnlyPermission(src, dest, srcQty);
+
+            if (quantityFactor != 1) {
+                (bestReserve, slippageRate) = kyberNetwork.findBestRateOnlyPermission(src, dest,
+                    (srcQty * quantityFactor));
+            } else {
+                slippageRate = expectedRate;
+            }
+        }
```

Comment:
regarding your PR comment.
I Agree expectedRateSmallQty could fail.
solution would be after this line:
uint ethQty = calcDestAmount(src, ETH_TOKEN_ADDRESS, srcQty, rateSrcToEth)

if ethQty == 0, can use ethqty = 1 or return 0.
both are fine from my side.

Reply:
Added!